packages/dds/sequence/src/test/sharedString.spec.ts

```diff
@@ -236,6 +236,29 @@ describe("SharedString", () => {
 			assert.equal(simpleMarker.properties?.color, "blue", "Could not annotate marker");
 		});
 
+		it("fails when the marker is updated", () => {
+			sharedString.insertText(0, "hello world");
+			// Insert a simple marker.
+			sharedString.insertMarker(6, ReferenceType.Simple, {
+				[reservedMarkerIdKey]: "markerId",
+			});
+			// Annotate the marker.
+			const props = { color: "blue" };
+			const simpleMarker = sharedString.getMarkerFromId("markerId") as Marker;
+			sharedString.annotateMarker(simpleMarker, props);
+			assert.equal(simpleMarker.properties?.color, "blue", "Could not annotate marker");
+			// Annotate the marker's ID.
+			const newIdProps = { [reservedMarkerIdKey]: "newIdValue" };
+			try {
+				sharedString.annotateMarker(simpleMarker, newIdProps);
+			} catch (error) {
+				assert(
+					error !== undefined,
+					"Error from attempting to update marker was not thrown",
+				);
```

Comment:
More of a nit, but I think `assert.equal` and `assert.notEqual` are preferrable when possible, because the failure message contains the values. When this assert fails, it will only print `false`, which ends up being more painful to debug.

```suggestion
				assert.notEqual(
					error,
					undefined,
					"Error from attempting to update marker was not thrown",
				);
```

I might also check more than just that the error is defined. Comparing against `error.message` or checking `instanceof` would be useful